tiledb/sm/array_schema/domain.h

```diff
@@ -489,7 +501,7 @@ class Domain {
   Layout cell_order_;
 
   /** The domain dimensions. */
-  std::vector<tdb_unique_ptr<Dimension>> dimensions_;
+  std::vector<std::shared_ptr<Dimension>> dimensions_;
```

Comment:
Why the change? I'm not opposed to it on principle, but it seems unnecessary here.

Reply:
The reason is that Dimension::deserialize returns shared_ptr of Dimension. shared_ptr<T> cannot be converted to unique_ptr<T>, as discussed in https://stackoverflow.com/questions/37884728/does-c11-unique-ptr-and-shared-ptr-able-to-convert-to-each-others-type